main.go

```diff
@@ -678,6 +688,39 @@ func (c *controller) saveHashring(ctx context.Context, hashring []receive.Hashri
 	return nil
 }
 
+func (c *controller) annotatePods(ctx context.Context, hashrings []receive.HashringConfig, statefulsets map[string]*appsv1.StatefulSet) {
+	for _, h := range hashrings {
+		if sts, exists := statefulsets[h.Hashring]; exists {
+			for i := 0; i < int(*sts.Spec.Replicas); i++ {
+				podName := fmt.Sprintf("%s-%d", sts.Name, i)
+
+				pod, err := c.klient.CoreV1().Pods(c.options.namespace).Get(ctx, podName, metav1.GetOptions{})
+				if err != nil {
+					level.Error(c.logger).Log("msg", "failed to get pod", "err", err)
+				}
+
+				buf, err := json.Marshal(h)
+				if err != nil {
+					level.Error(c.logger).Log("msg", "failed to marshal config", "err", err)
+				}
+
+				annotations := pod.Annotations
+				if annotations == nil {
+					annotations = make(map[string]string)
+				}
+
+				annotations["configHash"] = fmt.Sprintf("%f", hashAsMetricValue(buf))
```

Comment:
since this hash appears to already exist as a metric, perhaps a timestamp here would be useful so we can gauge when it was added?

Reply:
I believe that also sort of exists (`configmapLastSuccessfulChangeTime`), although it's probably a bit different than when the annotation was added... but either way I don't have strong preference here. We just need to annotate with something (part of the solution I'm not in love with 😉 ). I'll make this a timestamp, it probably makes more sense.